From fdf370ef90f799d2b5fed9e700c672566c00c2ba Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Fri, 15 Jan 2016 17:35:23 +0800
Subject: [PATCH 0865/5242] MLK-12271 Input: atkbd: add return value check
 after calling .dmi_check_system()

commit  212edc5e5f5658d90af292076e02dee337de8619 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add return value check after calling .dmi_check_system().
Reported by Coverity: CID18431

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/input/keyboard/atkbd.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/input/keyboard/atkbd.c b/drivers/input/keyboard/atkbd.c
index 7e75835..4511a17 100644
--- a/drivers/input/keyboard/atkbd.c
+++ b/drivers/input/keyboard/atkbd.c
@@ -1807,7 +1807,8 @@ static int __init atkbd_deactivate_fixup(const struct dmi_system_id *id)
 
 static int __init atkbd_init(void)
 {
-	dmi_check_system(atkbd_dmi_quirk_table);
+	if (!dmi_check_system(atkbd_dmi_quirk_table))
+		return -ENODEV;
 
 	return serio_register_driver(&atkbd_drv);
 }
-- 
1.7.9.5

